File: CS/15_Document/CreatePDFPortfolio.cs
using Spire.Pdf;
using Spire.Pdf.Annotations;
using Spire.Pdf.Collections;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CreatePDFPortfolio
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // The folder path for files
            string[] files = Directory.GetFiles(@"..\..\..\..\..\..\Data\CreatePDFPortfolio");

            // The target file path
            string targetFile = @"..\..\..\..\..\..\Data\Sample.pdf";

            // Create a PDF document with the target file path
            PdfDocument doc = new PdfDocument(targetFile);

            // Iterate through the files array and add each file to the document's collection
            for (int i = 0; i < files.Length; i++)
            {
                doc.Collection.Folders.AddFile(files[i]);

                // Create a subfolder named "SubFolder" followed by the current index
                PdfFolder folder = doc.Collection.Folders.CreateSubfolder("SubFolder" + (i + 1));

                // Add the current file to the subfolder
                folder.AddFile(files[i]);
            }

            // Save the document to a file named "CreatePDFPortfolio_out.pdf"
            String result = "CreatePDFPortfolio_out.pdf";
            doc.SaveToFile(result);

            // Dispose of the document to release any resources
            doc.Dispose();

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}

File: VB/15_Document/CreatePDFPortfolio.vb
Imports Spire.Pdf
Imports Spire.Pdf.Collections
Imports System.IO

Namespace CreatePDFPortfolio
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Get an array of file paths from the specified directory.
            Dim files() As String = Directory.GetFiles("..\..\..\..\..\..\Data\CreatePDFPortfolio")

            ' Specify the target file path for the PDF portfolio.
            Dim targetFile As String = "..\..\..\..\..\..\Data\Sample.pdf"

            ' Create a new PdfDocument object with the target file path.
            Dim doc As New PdfDocument(targetFile)

            ' Iterate through the files array and add each file to the document's collection.
            For i As Integer = 0 To files.Length - 1
                doc.Collection.Folders.AddFile(files(i))

                ' Create a subfolder for each file and add the file to it.
                Dim folder As PdfFolder = doc.Collection.Folders.CreateSubfolder("SubFolder" & (i + 1))
                folder.AddFile(files(i))
            Next i

            ' Specify the file name for the resulting PDF portfolio.
            Dim result As String = "CreatePDFPortfolio_out.pdf"

            ' Save the document as a PDF portfolio to the specified file.
            doc.SaveToFile(result)

            ' Close the document.
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)
        End Sub

        Private Sub PDFDocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class
End Namespace
